starport/services/network/networkchain/binarycache.go

```diff
@@ -0,0 +1,58 @@
+package networkchain
+
+import (
+	"strconv"
+
+	"github.com/tendermint/starport/starport/chainconfig"
+	"github.com/tendermint/starport/starport/pkg/checksum"
+	"github.com/tendermint/starport/starport/pkg/confile"
+	"github.com/tendermint/starport/starport/pkg/xfilepath"
+)
+
+const (
+	BinaryCacheDirectory = "binary-cache"
+	BinaryCacheFilename  = "list.yml"
+)
+
+type List struct {
+	CachedBinary map[string]string `json:"cached_binary" yaml:"cached_binary"`
+}
+
+// CacheBinaryForLaunchID caches hash sha256(sha256(binary) + sourcehash) for launch id
+func CacheBinaryForLaunchID(launchID uint64, binaryHash, sourceHash string) error {
+	cachePath, err := getBinaryCacheFilepath()
+	if err != nil {
+		return err
+	}
+	cacheList := List{CachedBinary: map[string]string{}}
```

Comment:
```suggestion
	var cacheList BinaryCacheList
```

We can use it like this if it works.